eodag/utils/s3.py

```diff
@@ -20,118 +20,449 @@
 
 logger = logging.getLogger("eodag.utils.s3")
 
+MIME_OCTET_STREAM = "application/octet-stream"
+
 
-def fetch(
-    bucket_name: str, key_name: str, start: int, len: int, client_s3: S3Client
+def fetch_range(
+    bucket_name: str, key_name: str, start: int, end: int, client_s3: S3Client
 ) -> bytes:
     """
     Range-fetches a S3 key.
 
     :param bucket_name: Bucket name of the object to fetch
     :param key_name: Key name of the object to fetch
-    :param start: Bucket name to fetch
-    :param len: Bucket name to fetch
+    :param start: Start byte position to fetch
+    :param end: End byte position to fetch
     :param client_s3: s3 client used to fetch the object
     :returns: Object bytes
     """
-    end = start + len - 1
-    s3_object = client_s3.get_object(
+    response = client_s3.get_object(
         Bucket=bucket_name, Key=key_name, Range="bytes=%d-%d" % (start, end)
     )
-    return s3_object["Body"].read()
-
-
-def parse_int(bytes: bytes) -> int:
-    """
-    Parses 2 or 4 little-endian bits into their corresponding integer value.
-
-    :param bytes: bytes to parse
-    :returns: parsed int
-    """
-    val = (bytes[0]) + ((bytes[1]) << 8)
-    if len(bytes) > 3:
-        val += ((bytes[2]) << 16) + ((bytes[3]) << 24)
-    return val
+    return response["Body"].read()
 
 
-def open_s3_zipped_object(
-    bucket_name: str, key_name: str, client_s3: S3Client, partial: bool = True
-) -> ZipFile:
+@dataclass
+class FileInfo:
     """
-    Open s3 zipped object, without downloading it.
-
-    See https://stackoverflow.com/questions/41789176/how-to-count-files-inside-zip-in-aws-s3-without-downloading-it;
-    Based on https://stackoverflow.com/questions/51351000/read-zip-files-from-s3-without-downloading-the-entire-file
-
-    :param bucket_name: Bucket name of the object to fetch
-    :param key_name: Key name of the object to fetch
-    :param client_s3: s3 client used to fetch the object
-    :param partial: fetch partial data if only content info is needed
-    :returns: List of files in zip
+    Describe a S3 object with basic f_info and its download state.
+
+    zip_filepath is the path inside the ZIP archive if the file is stored inside a ZIP.
+    data_start_offset is the offset in the ZIP archive where the file data starts.
+    file_start_offset is the offset in the logical (global) file stream where this file starts.
+    futures is a mapping of futures for each range (start byte -> future).
+    buffers is a mapping of start byte to data chunks that have been downloaded.
+    next_yield is the next offset to yield in the file, used to track progress during
+    downloading and yielding chunks.
+    rel_path is the relative path of the file, if applicable (e.g., inside a ZIP archive).
+    data_type is the MIME type of the file, defaulting to application/octet-stream
+    if not guessed. It can be updated based on the file extension or content type.
```

Comment:
please format documentation (see https://eodag--1740.org.readthedocs.build/en/1740/api_reference/utils.html#eodag.utils.s3.FileInfo).

Use per-parameter documented comments like in `PluginConfig`.